mmv1/templates/terraform/examples/apigee_nat_address_basic.tf.erb

```diff
@@ -0,0 +1,75 @@
+data "google_client_config" "current" {}
+
+resource "google_compute_network" "apigee_network" {
+  name = "apigee-network"
+}
+
+resource "google_compute_global_address" "apigee_range" {
+  name          = "apigee-range"
+  purpose       = "VPC_PEERING"
+  address_type  = "INTERNAL"
+  prefix_length = 21
+  network       = google_compute_network.apigee_network.id
+}
+
+resource "google_service_networking_connection" "apigee_vpc_connection" {
+  network                 = google_compute_network.apigee_network.id
+  service                 = "servicenetworking.googleapis.com"
+  reserved_peering_ranges = [google_compute_global_address.apigee_range.name]
+}
+
+resource "google_kms_key_ring" "apigee_keyring" {
+  name     = "apigee-keyring"
+  location = "us-central1"
+}
+
+resource "google_kms_crypto_key" "apigee_key" {
+  name            = "apigee-key"
+  key_ring        = google_kms_key_ring.apigee_keyring.id
+
+  lifecycle {
+    prevent_destroy = true
+  }
+}
+
+resource "google_project_service_identity" "apigee_sa" {
+  provider = google-beta
+  project  = google_project.project.project_id
+  service  = google_project_service.apigee.service
+}
+
+resource "google_kms_crypto_key_iam_binding" "apigee_sa_keyuser" {
+  crypto_key_id = google_kms_crypto_key.apigee_key.id
+  role          = "roles/cloudkms.cryptoKeyEncrypterDecrypter"
+
+  members = [
+    "serviceAccount:${google_project_service_identity.apigee_sa.email}",
+  ]
+}
+
+resource "google_apigee_organization" "apigee_org" {
+  analytics_region                     = "us-central1"
+  display_name                         = "apigee-org"
+  description                          = "Terraform-provisioned Apigee Org."
+  project_id                           = data.google_client_config.current.project
+  authorized_network                   = google_compute_network.apigee_network.id
+  runtime_database_encryption_key_name = google_kms_crypto_key.apigee_key.id
+
+  depends_on = [
+    google_service_networking_connection.apigee_vpc_connection,
+    google_kms_crypto_key_iam_binding.apigee_sa_keyuser,
+  ]
+}
+
+resource "google_apigee_instance" "apigee_instance" {
+  name                     = "tf-test%{random_suffix}"
+  location                 = "us-central1"
+  description              = "Terraform-managed Apigee Runtime Instance"
+  display_name             = "tf-test%{random_suffix}"
+  org_id                   = google_apigee_organization.apigee_org.id
+  disk_encryption_key_name = google_kms_crypto_key.apigee_key.id
+}
+
+resource "google_apigee_nat_address" "apigee-nat" {
+  instance_id = google_apigee_instance.apigee_instance.id
+}
```

Comment:
Could you please add the `name` argument to this resource? Sorry, I should have picked up on this before!

When I ran the acceptance test (that is generated from this example file) I edited the code in the way shown below, and that allowed the test to run and pass

```diff
resource "google_apigee_nat_address" "apigee_nat_address" {
+  name        = "tf-test%{random_suffix}"
  instance_id = google_apigee_instance.apigee_instance.id
}
```

Reply:
Ah I missed that part. I assumed the `"tf-test%{random_suffix}"` should only show up on the resource you want to test, right? Added a commit to address this.

Reply:
It depends on the test, but usually we give most resources names with `random_suffix` appended at the end. This helps avoid a situation where a test with hardcoded names and fails on run # 1, leaving resources in GCP with that name, and later on run # 2 the test fails because it wants to create a resource that already exists with that name.

Having said that though, this test makes all the resources in a new project (and then deletes the project at the end) so hardcoded names are ok here!